olmeiro-orozco/tarea-5/js/juego.js

```diff
@@ -0,0 +1,72 @@
+import { Caja } from "./caja.js";
+import { Estado } from "./estado.js";
+import { cambiarCajas, puzzleResuelto } from "./helpers.js";
+
+export class Juego {
+  estado;
+
+  constructor(estado) {
+    this.estado = estado;
+    this.renderizarJuego();
+  }
+
+  static iniciar() {
+    return new Juego(Estado.iniciar());
+  }
+
+  setEstado(nuevoEstado) {
+    this.estado = { ...this.estado, ...nuevoEstado };
+    this.renderizarJuego();
+  }
+
+  clickEnCaja = (caja) => {
+    return () => {
+      const cajasAlrededor = caja.getCajasAlrededor();
+      const cajaBlanca = cajasAlrededor.find(
+        (cajaAlrededor) =>
+          this.estado.grid[cajaAlrededor.fila][cajaAlrededor.columna] === 0
+      );
+
+      if (cajaBlanca) {
+        const nuevaGrid = [...this.estado.grid];
+        cambiarCajas(nuevaGrid, caja, cajaBlanca);
+
+        if (puzzleResuelto(nuevaGrid)) {
+          this.setEstado({
+            estado: "ganaste",
+            grid: nuevaGrid,
+          });
+        } else {
+          this.setEstado({
+            grid: nuevaGrid,
+          });
+        }
+      }
+    };
+  };
+
+  renderizarJuego() {
+    const { grid, estado } = this.estado;
+    const nuevaGrid = document.createElement("section");
+
+    nuevaGrid.className = "grilla";
+
+    for (let i = 0; i < 4; i++) {
```

Comment:
@OlmeiroOrozco , you should use modern approach like map, forEach or reduce, also remember to name properly the variable i and j